azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/TaskWrapper.kt

```diff
@@ -20,14 +20,18 @@ internal class TaskWrapper<
 
     fun <S : BuildableScenario<AF>> run(scenario: S): TaskResult<S, R> {
         val start = System.nanoTime()
-        val outCapture = LogAndCaptureOutputStream { Log.info(it) }
-        val errCapture = LogAndCaptureOutputStream { Log.error(it) }
+        val redirectStdStreams = System.getProperty(redirectStdStreamsPropertyName)?.toBoolean() == true
+        val out = System.out
+        val err = System.err
+        val outCapture = LogAndCaptureOutputStream { if (redirectStdStreams) Log.info(it) else out.println(it) }
+        val errCapture = LogAndCaptureOutputStream { if (redirectStdStreams) Log.error(it) else err.println(it) }
```

Comment:
It might just be a tiny optimisation, but I would prefer if the check was done out of the lambdas, when setting the values, i.e. something like
```suggestion
        if (redirectStdStreams) {
            val outCapture = LogAndCaptureOutputStream { Log.info(it) }
            val errCapture = LogAndCaptureOutputStream { Log.error(it) }
        } else {
            val outCapture = LogAndCaptureOutputStream { out.println(it) }
            val errCapture = LogAndCaptureOutputStream { err.println(it) }
        }
```

Reply:
that is not an optimization as hotspot will eliminate the difference on like the 2nd loop and what you have can't work unless you have two if-blocks or use vars for the captures; this to me was the least worst option